zerver/views/message_fetch.py

```diff
@@ -1102,8 +1103,10 @@ def get_messages_backend(
     else:
         for row in rows:
             message_id = row[0]
-            flags = row[1]
-            user_message_flags[message_id] = UserMessage.flags_list_for_flags(flags)
+            um_rows = UserMessage.objects.filter(user_profile=user_profile, message__id=message_id)
+            user_message_flags[message_id] = []
+            for um in um_rows:
+                user_message_flags[message_id] += um.flags_list()
```

Comment:
Can you explain the thinking behind this change?  It seems like a significant performance regression, in that we're now doing a separate `UserMessage` query that we weren't doing before in this case.